stumpy/core.py

```diff
@@ -121,7 +121,8 @@ def norm_func(Q, T, A_norm=None, other_norm=None, normalize=True, p=2.0):
         The desired z-normalized/non-normalized function (or class)
     """
     if exclude is None:
-        exclude = ["normalize", "p"]
+        exclude = ["normalize", "p", "k"]  # remove "k" after updating
```

Comment:
Could you please elaborate what you meant by this `can be used in non-top-k scenarios`?

Reply:
The problem is that a function may have `normalize`, `p`, and `k` parameters but the `k` may not be referring to "top-k". The perfect example of this is the `snippets` function. In that case, we should NOT exclude `k` when `exclude=None`!

Does that make sense?

Reply:
I got it!! If I remember correctly, I got an error in `test_non_normalized_decorator.py` because `k` was not defined in non-normalized versions of `stump`. So, what I can do for now is to just add parameter `k=1` to non-normalized version of `stump`. Then, after taking care of `stump`, we can take care of its non-normalized version completely.